### PR TITLE
Move components to ts

### PR DESCRIPTION
**Pull request by Winner95**
It should resolve issue #14

### CODE REVIEW

File: src/components/section/section.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            'styleguide-section': true,
+            vrt: isVrtEnabled,
+        })
+    };
+
+    return (
+        <section className={classnames.block}>
+            <div className="styleguide-section__content">
+                {list.filter(Boolean).map(item => {
+                    return (
```
Comment (kdzuin) [inline src/components/section/section.tsx]: no need in this `return {`, just do `item => (`

File: src/components/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -2,12 +2,17 @@ import React from 'react';
...
+    children: React.ReactNode[];
+}
+
+class Sidebar extends React.Component<SidebarProps> {
+    constructor(props: SidebarProps) {
+        super(props);
     }
 
     render() {
+        const { children } = this.props;
```
Comment (kdzuin) [inline src/components/sidebar/sidebar.tsx]: do we need to destruct in sake of a single prop being used once?

File: src/components/component/component.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import cx from 'classnames';
+
+import Sandbox from '../sandbox/sandbox';
+
+import './component.scss';
+
+interface TestProps {
+    name?: string,
+    Component?: any
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/component/component.tsx]: let's keep for now it is as 'any' - it need some deeper research.

File: src/components/error-boundary/error-boundary.tsx [hunk 4]
```diff
@@ -2,8 +2,22 @@ import React from 'react';
 
 import './error-boundary.scss';
 
-export default class ErrorBoundary extends React.Component {
-    constructor(props) {
+interface ErrorBoundaryProps {
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/error-boundary/error-boundary.tsx]: updated definition

File: src/components/error-boundary/error-boundary.tsx [hunk 5]
```diff
@@ -2,8 +2,22 @@ import React from 'react';
...
+
+};
+
+interface ErrorBoundaryErrorInfo {
+    componentStack: Object | string | null;
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/error-boundary/error-boundary.tsx]: yes, it is. For consistency changed it to {}

File: src/components/error-boundary/error-boundary.tsx [hunk 6]
```diff
@@ -2,8 +2,22 @@ import React from 'react';
...
+}
+
+interface ErrorBoundaryState {
+    hasError: boolean;
+    error: Object | null;
+    errorInfo: ErrorBoundaryErrorInfo | null;
+};
+
+export default class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/error-boundary/error-boundary.tsx]: adde children, which are needed there.

File: src/components/dialog/dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+import './dialog.scss';
+
+const KEYCODES = {
+    ESCAPE: 27,
+};
+
+interface DialogProps {
+    isOpened?: boolean;
+    onClose?: Function;
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/dialog/dialog.tsx]: updated

File: src/components/search-field/search-field.scss [hunk 8]
```diff
@@ -42,10 +42,14 @@ $ui-border-color: #ccc;
...
-    left: 20px;
     top: 50%;
+    left: 20px;
+    display: flex;
     transform: translateY(-50%);
-    color: $ui-border-color;
-    width: 24px;
-    height: 24px;
+
+    .styleguide-icon {
```
[the remark this answers is not in the document]
Comment (Winner95) [reply src/components/search-field/search-field.scss]: yes, fixed by adding size prop

File: src/components/error-boundary/error-boundary.tsx [hunk 9]
```diff
@@ -2,8 +2,22 @@ import React from 'react';
 
 import './error-boundary.scss';
 
-export default class ErrorBoundary extends React.Component {
-    constructor(props) {
+interface ErrorBoundaryProps {
+    children: React.ReactNode[];
```
Comment (kdzuin) [inline src/components/error-boundary/error-boundary.tsx]: no need to define `React.ReactNode[]`, the type itself can be an array.

File: src/components/error-boundary/error-boundary.tsx [hunk 10]
```diff
@@ -13,7 +27,7 @@ export default class ErrorBoundary extends React.Component {
         };
     }
 
-    componentDidCatch(error, errorInfo) {
+    componentDidCatch(error: object | null, errorInfo: ErrorBoundaryErrorInfo) {
```
Comment (kdzuin) [inline src/components/error-boundary/error-boundary.tsx]: We can point to a part of interface, like `ErrorBoundaryState['error']`

Comment (kdzuin) [reply src/components/error-boundary/error-boundary.tsx]: or we can define the type out of the interface and point it to the defined type for error `ErrorType`

File: src/components/sidebar/sidebar.tsx [hunk 11]
```diff
@@ -2,9 +2,13 @@ import React from 'react';
 
 import './sidebar.scss';
 
-class Sidebar extends React.Component {
-    constructor() {
-        super();
+interface SidebarProps {
+    children: React.ReactNode[];
```
Comment (kdzuin) [inline src/components/sidebar/sidebar.tsx]: no need in `ReactNode[]`,
```
    type ReactText = string | number;
    type ReactChild = ReactElement | ReactText;

    interface ReactNodeArray extends Array<ReactNode> {}
    type ReactFragment = {} | ReactNodeArray;
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```
see `ReactFragment` and `ReactNodeArray`

File: tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+                "*",
+                "src/components/*",
+                "./node_modules/*",
+            ]
+        }
+    },
+    "include": ["./**/*"],
+    "exclude": [
+        "node_modules",
+        "examples/typescript/src/*",
```
Comment (kdzuin) [inline tsconfig.json]: maybe we can exclude the whole `examples` folder?

Comment (Winner95) [reply tsconfig.json]: yes, it make sense, fixed it

File: src/components/sidebar/sidebar.tsx (hunk 11 above)
Comment (Winner95) [reply src/components/sidebar/sidebar.tsx]: I agree, I see your point

File: src/components/error-boundary/error-boundary.tsx (hunk 10 above)
Comment (Winner95) [reply src/components/error-boundary/error-boundary.tsx]: makes sense, updated in this PR

**Pull request merged**